nselib/httpcookies.lua

```diff
@@ -176,6 +176,16 @@ CookieJar = {
   -- sends a cookiejar taking only the eligible cookies into consideration
   check_cookie_attributes = function(self, host, port, path)
     local cookie_table = {}
+    --Changing type of port to table cause shortport library expects a port table
+    if type(port) == 'number' then
+      port = {port}
```

Comment:
I believe this is wrong because you are not creating the port table correctly. In your if clause you know its a number yet you access a member (port.number) so that's probably why it is not working as expected. Make sure the table is created before accessing 'number'.